### ISSUE TITLE
onNotification not called for localNotifications

### ISSUE THREAD

**Issue by Faolain** [zo0r/react-native-push-notification#1024]
The onNotification callback only ever fires when receiving remote Notifications(in foreground) and not when receiving local Notifications(in foreground or background). I can make the onNotification trigger by tapping the localNotification and opening it but I need it to fire without opening as I have some logic which needs this. Has anyone else experienced this? Should localNotifications trigger the onNotification callback? If so why do remote(fcm) notifications trigger it but not local? Any help is greatly appreciated.

**Versions**
```
Android 5 and Android 6
"react-native": "0.57.0"
"react-native-push-notification": "^3.1.2"
```

**Push Notification Configuration**
```const configure = store => {
  PushNotification.configure({
    onRegister({ token, os }) {
      // process token
      store.dispatch(registerPushToken({ token, os }));
    },

    onNotification(notification) {
      console.log(notification)
    },

    // ANDROID ONLY: GCM or FCM Sender ID (product_number) (optional -
    // not required for local notifications, but is need to receive remote push notifications)
    senderID: ''xxx",

    permissions: {
      alert: true,
      badge: true,
      sound: true,
    },
    popInitialNotification: false,
    requestPermissions: false,
  });
};

**Comment by polsson12** [comment]
I'm facing the same issue, have you been able to make any progess on this?

**Comment by wickstjo** [comment]
Same issue. I want to pass the notification data through a function before prompting it, but I can't find a way to listen for a trigger and it would make sense for onNotification method to be able to do that.

**Comment by NikhilP99** [comment]
Same issue.

**Comment by ThalKod** [comment]
Same issue, does anyone have a fix ?

**Comment by liplylie** [comment]
Same issue

**Comment by AfrazHussain** [comment]
Having the same issue here. Did anyone manage to figure this out?

**Comment by ThalKod** [comment]
@AfrazHussain Yes, but by switching to [react-native-firebase](https://rnfirebase.io/docs/v5.x.x/notifications/introduction) using the Notifications module.

**Comment by Adr1ann** [comment]
Any solution here?

**Comment by xaviergerard** [comment]
Same issue for me.  Who's got a solution?

**Comment by Sanan4li** [comment]
same here... I want to open a specific screen when user clicks on action but can't find a way to do that

**Comment by indigomelody** [comment]
Same here! Any help on this at all would be really appreciated!

**Comment by abdymm** [comment]
same here, any solution for this?
